test/transformers/test_tvd.py

```diff
@@ -0,0 +1,132 @@
+from test.utils import supports_bfloat16
+
+import pytest
+import torch
+
+from liger_kernel.transformers.tvd import LigerTVDLoss
+
+
+class TorchTVDLoss(torch.nn.Module):
+    def __init__(self, reduction="batchmean"):
+        super(TorchTVDLoss, self).__init__()
+        self.reduction = reduction
+
+    def forward(self, p, q):
+
+        tvd = torch.abs(p - q) / 2.0
+
+        if self.reduction == "mean":
+            return torch.sum(tvd) / (p.size(0) * p.size(1))
+        elif self.reduction == "sum":
+            return torch.sum(tvd)
+        elif self.reduction == "none":
+            return tvd
+        elif self.reduction == "batchmean":
+            return torch.sum(tvd) / p.size(0)
+        else:
+            raise ValueError("Invalid reduction type.")
+
+
+_SHAPE_PARAMS = (
+    "B, T, V",
+    [
+        (1, 4096, 32000),
+        (32, 4096, 1024),
+        (41, 401, 1271),
+        pytest.param(
+            1,
+            4096,
+            128256,
+            marks=pytest.mark.skipif(
+                torch.cuda.get_device_properties(0).total_memory
+                < 36 * 1000 * 1000 * 1000,
+                reason="This test requires a GPU with at least 36GB of memory",
+            ),
+        ),
+        (3, 423, 32000),
+    ],
+)
+
+_DTYPE_PARAMS = (
+    "dtype, atol, rtol",
+    [
+        pytest.param(
+            torch.bfloat16,
+            1e-8,
+            5e-2,
```

Comment:
could you help to experiment what is the lowest `rtol` that will not fail this test for bf16? Thanks!

Reply:
@yundai424  approx 5.2e-18 is the lowest rtol that does not fail.
`Test passed with rtol=5.0e-02
Testing with rtol=4.5e-02...
Test passed with rtol=4.5e-02
Testing with rtol=4.0e-02...
Test passed with rtol=4.0e-02
Testing with rtol=3.5e-02...
Test passed with rtol=3.5e-02
Testing with rtol=3.0e-02...
Test passed with rtol=3.0e-02
Testing with rtol=2.5e-02...
Test passed with rtol=2.5e-02
Testing with rtol=2.0e-02...
Test passed with rtol=2.0e-02
Testing with rtol=1.5e-02...
Test passed with rtol=1.5e-02
Testing with rtol=1.0e-02...
Test passed with rtol=1.0e-02
Testing with rtol=5.0e-03...
Test passed with rtol=5.0e-03
Testing with rtol=5.2e-18...
Test passed with rtol=5.2e-18
Testing with rtol=-5.0e-03...
FAILED`

Reply:
@yundai424 Please ignore previous comment. 1e-6 is the lowest rtol which is not failing.